### DISCUSSION TITLE
can not init   `> Downloading Prisma engines for Node-API for windows [                    ] 0%Error: request to https://binaries.prisma.sh/all_commits/aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a/windows/query_engine.dll.node.gz.sha256 failed, reason: read ECONNRESET`

### DISCUSSION THREAD

**Discussion by jeffhuien** [prisma/prisma#21938] [Q&A]
### Bug description

can not init

### How to reproduce

1. setup  ``` pnpm add prisma  @prisma/client```

6. Run `pnpm prisma init` or  `npx prisma init`
7. See error

linux
`> Downloading Prisma engines for Node-API for debian-openssl-1.1.x [                    ] 0%Error: request to https://binaries.prisma.sh/all_commits/aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a/debian-openssl-1.1.x/libquery_engine.so.node.gz.sha256 failed, reason:`
windows
`> Downloading Prisma engines for Node-API for windows [                    ] 0%Error: request to https://binaries.prisma.sh/all_commits/aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a/windows/query_engine.dll.node.gz.sha256 failed, reason: read ECONNRESET`

### Prisma information

```prisma
// Add your schema.prisma
0
```

```ts

0
// Add your code using Prisma Client
```

### Environment & setup

- OS: Windows, ubuntu
- Database: mysql 
- Node.js version: v21.1.0 ,  v20.9.0

### Prisma Version

```
prisma@5.5.2

 pnpm prisma -v
> Downloading Prisma engines for Node-API for debian-openssl-1.1.x [                    ] 0%Error: request to https://binaries.prisma.sh/all_commits/aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a/debian-openssl-1.1.x/schema-engine.gz.sha256 failed, reason:
```

**Comment by ludralph** [comment]
Hi @jeffhuien 👋 

Thank you for raising this question.

Can you access the URL of the Prisma binaries manually in a browser on your machine? For example, for debian-openssl-1.1.x, the URL would be https://binaries.prisma.sh/all_commits/aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a/debian-openssl-1.1.x/libquery_engine.so.node.gz.sha256.

Also, your internet connection might be blocking the servers Prisma is using. They are hosted on Cloudflare, which is a very common CDN. You could try a different internet connection, mobile hotspot, or VPN to see if this solves the problem.

**Reply by jeffhuien** [reply]
Yes it can download the file in the link manually but even if I use a VPN it still doesn't work

the file content like this
a115da6284cf291a34e443e7efc5114d59d45d9a89fe9811a02dc049bf80f7b1  schema-engine.exe

**Reply by xxlbb123** [reply]
I also encountered the same problem, manually setting vpn is not effective, but I can use vpn to visit the website to download files, but where should the downloaded files be placed? @ludralph

**Comment by hooper-hc** [comment]
I also encountered the same problem. I executed curl on my personal computer: https://binaries.prisma.sh/all_commits/aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a/debian-openssl-1.1.x/libquery_engine.so.node.gz.sha256, and I could get the return normally. 

But on the cloud server , it often fails, error message:

xx-service: > prisma db pull && prisma generate
xx-service:
xx-service: Error: request to https://binaries.prisma.sh/all_commits/12e25d8d06f6ea5a0252864dd9a03b1bb51f3022/linux-musl-openssl-3.0.x/schema-engine.gz failed, reason: read ECONNRESET
xx-service:  ELIFECYCLE  Command failed with exit code 1.
xx-service:  ELIFECYCLE  Command failed with exit code 1.
xx-service:  ELIFECYCLE  Command failed with exit code 1.
xx-service: ERROR: command finished with error: command (/builds/0/reddoc-suite/apps-service/common-suite-service) /usr/local/bin/pnpm run build:test exited (1)

**Comment by wuhao199368** [comment]
I've encountered the same problem and solve it accidently. The way to solve this problem is so weird. 

Firstly here's my error message: 
"Error: request to https://binaries.prisma.sh/all_commits/69d742ee20b815d88e17e54db4a2a7a3b30324e3/windows/schema-engine.exe.sha256 failed, reason: read ECONNRESET", and here's the step how I initiate Prisma successfully:
1) Disable my VPN setting;
2) Copy and paste the above URL in my browser and make sure that I can access the URL(Make sure the you paste your own URL from the error message you encountered); If you can get access to the URL, just don't turn off the corresponding page, keep it alive;
3) Re-run the "npx prisma init" command.

I found that I can download the Prisma engines super fast after I do all the procedures above, but I don't know why it works. Maybe it's just lucky for me. You can try it yourself.
![微信图片_20240913233603](https://github.com/user-attachments/assets/f9b62789-96fe-454d-ba5e-f63572709599)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
